### PR TITLE
feat(rxModalAction): add dismiss-hook support

### PR DESCRIPTION
**Pull request by CITguy**
JIRA: https://jira.rax.io/browse/FRMW-59

- [x] Dev LGTM
- [x] Dev LGTM
- [x] Design LGTM
- [ ] QE LGTM

### Documentation (BEFORE)
20150814-page-before

### Documentation (AFTER)
20150820-page-after

### After Dismiss (Cancel/Close)
screen shot 2015-08-14 at 1 38 51 pm

### After Submit
screen shot 2015-08-14 at 1 39 09 pm

### CODE REVIEW

File: src/rxModalAction/README.md [hunk 1]
```diff
@@ -10,17 +10,30 @@ Due to the complex nature of the information passed into modal windows, HTML is
...
+## Hooks
 
-`rxModalAction` allows you to take actions before and after the modal window is shown. They are optional, and the modal window is fully functional without either being defined. Both are passed in as functions that are called on open and close of the modal window.
+`rxModalAction` allows you to define functions to be called at different points of the modal lifecycle.
+These hooks are optional and the modal window is fully functional without any being defined.
 
 ### Pre-hook
 
-Use a `pre-hook` to populate field information inside of the modal. This is useful when you have information you don't want loaded when the page is first opened, but do need for the modal. It's also useful for dynamic information that's based on actions taken
+The `pre-hook` function is called before the model window is shown.  Use a this to populate field information inside of the modal.
```
Comment (Droogans) [inline src/rxModalAction/README.md]: > *modal* window is shown

Comment (CITguy) [inline src/rxModalAction/README.md]: ah, good catch. will update

File: src/rxModalAction/README.md [hunk 2]
```diff
@@ -10,17 +10,30 @@ Due to the complex nature of the information passed into modal windows, HTML is
...
+The `post-hook` function is called after the modal is submitted.  A `post-hook` is useful to take actions based upon input in the modal.
+For example, you can use the user input that gets entered to send API requests with specific JSON data.
+Or you can simply run a pre-defined API call (assuming the modal is a simple confirmation dialog).
+
+### Dismiss-hook
+
+The `dismiss-hook` function is called after the modal is closed without submitting.  This is useful to make changes to UI state when the user wants to cancel the action.
+For example, you may use this to return an indeterminate component to a previous state (e.g. toggle switches).
+
+**NOTE:** This hook will not be called when using a custom controller.  If you want similar functionality with a custom controller, you'll want to put your logic within your `$scope.cancel()` function.
```
Comment (megalord) [inline src/rxModalAction/README.md]: I don't think this is accurate.  The hook *will* be called when using a custom controller as long as `$modalInstance.dismiss()` is called, right?

Comment (CITguy) [inline src/rxModalAction/README.md]: Yeah... :disappointed: That's correct. This was an afterthought of an experimental implementation that I thought I had implemented. I'll get this updated.

File: src/rxModalAction/README.md [hunk 3]
```diff
@@ -10,17 +10,28 @@ Due to the complex nature of the information passed into modal windows, HTML is
...
-A `post-hook` is useful to take actions based upon input in the modal. For example, you can use the user input that gets entered to send API requests with specific JSON data. Or you can simply run a pre-defined API call (assuming the modal is a simple confirmation dialog).
+The `post-hook` function is called after the modal is submitted.  A `post-hook` is useful to take actions based upon input in the modal.
+For example, you can use the user input that gets entered to send API requests with specific JSON data.
+Or you can simply run a pre-defined API call (assuming the modal is a simple confirmation dialog).
+
+### Dismiss-hook
+
+The `dismiss-hook` function is called after the modal is closed without submitting (i.e. after `$modalInstance.dismiss()` is called).
+This is useful to make changes to UI state when the user wants to cancel the action.
+For example, you may use this to return an indeterminate component to a previous state (e.g. toggle switches).
```
Comment (parlarjb) [inline src/rxModalAction/README.md]: I would add an explicit point here that this covers hitting the "Cancel" button or the "X" dismiss button

**Comment by parlarjb** [comment]
LGTM

**Comment by glynnis** [comment]
LGTM

File: src/rxModalAction/README.md [hunk 4]
```diff
@@ -10,17 +10,28 @@ Due to the complex nature of the information passed into modal windows, HTML is
...
 ### Post-hook
 
-A `post-hook` is useful to take actions based upon input in the modal. For example, you can use the user input that gets entered to send API requests with specific JSON data. Or you can simply run a pre-defined API call (assuming the modal is a simple confirmation dialog).
+The `post-hook` function is called after the modal is submitted.  A `post-hook` is useful to take actions based upon input in the modal.
+For example, you can use the user input that gets entered to send API requests with specific JSON data.
+Or you can simply run a pre-defined API call (assuming the modal is a simple confirmation dialog).
+
+### Dismiss-hook
+
+The `dismiss-hook` function is called after the modal is closed without submitting. This may happen by clicking the "Cancel" button, clicking the "X" button in the top right, or explicitly calling `$modalInstance.dismiss()` in your javascript logic.
```
Comment (megalord) [inline src/rxModalAction/README.md]: The `dismiss-hook` is also called when the escape key is pressed.  It's implied by the first sentence, but you have gone to the trouble of listing all the other possibilities, so I figure it's worth including.

**Comment by megalord** [comment]
LGTM

File: src/rxModalAction/README.md (hunk 4 above)
Comment (CITguy) [inline src/rxModalAction/README.md]: Done.

**Comment by CITguy** [comment]
* changed list of dismiss scenarios to a bulleted list
* updated screenshot

**Pull request merged**